src/textual/app.py

```diff
@@ -650,6 +652,30 @@ def __init__(
         # Size of previous inline update
         self._previous_inline_height: int | None = None
 
+    def __init_subclass__(cls, *args, **kwargs) -> None:
+        for variable_name, screen_collection in (
+            ("SCREENS", cls.SCREENS),
+            ("MODES", cls.MODES),
+        ):
+            for screen_name, screen_object in screen_collection.items():
+                if not isinstance(screen_name, str):
```

Comment:
I think we can do without this check. It's not likely the dev would make this error. If we checked all typing at runtime, then the codebase would be full of such checks.

Kudos for thinking of it tthough!